src/www/infrastructure/memory_storage.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export class InMemoryStorage implements Storage {
+  readonly length = 0;
+  [key: string]: {};
```

Comment:
I don't think you should use [] on Storage. That conflicts with the object members.

Reply:
I believe we must in order to implement the [Storage interface](https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L15394).